### PR TITLE
Add Basic Court UI

### LINKED ISSUE #1: Add Basic Court
- Add Court Page
- Add names display

### PR DESCRIPTION
**Pull request by victor77dev**
### Description
- Add Court Page
- Add names display
- Add typescript check

### QA
- [ ] Chrome

### Issuer
- https://github.com/victor77dev/badminton-scoreboard/issues/1

### TODO
- [ ] any follow-up here that is out of scope

### CODE REVIEW

File: src/features/court/index.jsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+
+import {HalfCourt} from './components/half-court';
+import {SubCourt} from './components/sub-court';
```
Comment (kavimaluskam) [inline src/features/court/index.jsx]: space between `{` `}` is not necessary in your lint ?

Comment (victor77dev) [reply src/features/court/index.jsx]: yes, it's not. But I suggest I should make it consistent.

File: src/App.js [hunk 2]
```diff
@@ -1,24 +1,12 @@
-import logo from './logo.svg';
-import './App.css';
+import React, { Fragment } from 'react';
+import { CourtPage } from 'pages/court';
```
Comment (kavimaluskam) [inline src/App.js:2]: just curious, any config setting is required for using absolute path import?
and if you are vscode, can you quickly open this file by `Ctrl + click`?

File: src/features/court/index.jsx (hunk 1 above)
Comment (victor77dev) [reply src/features/court/index.jsx]: BTW, my team uses no space convention.
But let's use space in this repo.

File: src/App.js (hunk 2 above)
Comment (victor77dev) [reply src/App.js:2]: https://github.com/victor77dev/badminton-scoreboard/pull/5/files#diff-b55cdbef4907b7045f32cc5360d48d262cca5f94062e353089f189f4460039e0R14
`baseUrl` in tsconfig.json is the path for it.

> can you quickly open this file by Ctrl + click?

Yes

Comment (kavimaluskam) [reply src/App.js:2]: i need a plugin to do global import in gatsby as well, and i cannot quick open publicly imported file lol

Nice to see your configuration

File: src/features/court/components/sub-court/index.module.scss [hunk 3]
```diff
@@ -0,0 +1,21 @@
+.container {
+  height: 100%;
+  width: 100%;
+}
+
+.court_image {
```
Comment (victor77dev) [inline src/features/court/components/sub-court/index.module.scss]: @kavimaluskam how do you usually name the css class? Which format you use?

Comment (kavimaluskam) [reply src/features/court/components/sub-court/index.module.scss]: i use `-` over `_`.

for naming ideas, usually its like the react module itself (i.e. `HalfCourt`) or
some styling with the module (`GreenCourt`, `FatDom`) ?

Comment (victor77dev) [reply src/features/court/components/sub-court/index.module.scss]: http://getbem.com/naming/
i think we should follow bem naming for css.

A blog post with more examples
https://www.freecodecamp.org/news/css-naming-conventions-that-will-save-you-hours-of-debugging-35cea737d849/

Comment (victor77dev) [reply src/features/court/components/sub-court/index.module.scss]: Screenshot 2020-12-30 at 19 30 13
I think the lib is converting scss into the naming pattern. Also, I couldn't import and have variable `styles.sub-court`.

@kavimaluskam check the update make sense to you?

Comment (kavimaluskam) [reply src/features/court/components/sub-court/index.module.scss]: I wonder the name would be too generic 🤔 , but it looks OK as long as the naming is only used in the `sub-court` or `half-court` module.

Comment (kavimaluskam) [reply src/features/court/components/sub-court/index.module.scss]: It follows my naming practice as in styled-component. 

But not sure how to do when we have some global css/class naming needs.

Comment (victor77dev) [reply src/features/court/components/sub-court/index.module.scss]: > I wonder the name would be too generic

I think that's why it's put the block part name in front and then the sub name

**Review by kavimaluskam — APPROVED** [review]

File: src/features/court/index.jsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      bodyStyle.setProperty('--court_height', `${height}px`);
+    }
+
+    onScreenChange();
+    window.addEventListener('resize', onScreenChange);
+
+    return () => {
+      window.removeEventListener('resize', onScreenChange);
+    };
+  });
```
Comment (kavimaluskam) [inline src/features/court/index.jsx:25]: I used `useEffect` and event listener for some styling effect as well, but the animation / response is not much smooth than pure CSS. Do you experience similarly?

File: src/features/court/index.module.scss [hunk 5]
```diff
@@ -0,0 +1,21 @@
+.court {
+  display: flex;
+  width: var(--court_width);
+  height: var(--court_height);
```
Comment (kavimaluskam) [inline src/features/court/index.module.scss:4]: `var` for css is a cool tricks 🤣

File: src/features/court/index.jsx (hunk 4 above)
Comment (victor77dev) [reply src/features/court/index.jsx:25]: @kavimaluskam no idea🤔
It looks fine for me.
What do you mean by pure CSS vs `useEffect` one?
Is there an example?

**Pull request merged**